sketch.js

```diff
@@ -12,64 +18,80 @@ for (var i=0; i <= 16; i++){
   var yPosition = 550;
   var rectWidth = 5;
   var datayear= -data[i]/120;
-  
+
   var check = collidePointCircle(mouseX,mouseY,153, 470, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
-		text("90 Million",130,450);
+		words["text"] = "90 Million";
+    words["x"] = 130;
+    words["y"] = 450;
 		noStroke;
 	}
-	
+
 	var check = collidePointCircle(mouseX,mouseY, 176, 460, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("100 Million",150,450);
+		words["text"] = "100 Million";
+    words["x"] = 150;
+    words["y"] = 450;
 	}
-		
+
 	var check = collidePointCircle(mouseX,mouseY, 268, 430, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("130 Million",245,415);
+		words["text"] = "130 Million";
+    words["x"] = 245;
+    words["y"] = 415;
 	}
-	 
-	
+
+
 	var check = collidePointCircle(mouseX,mouseY, 337, 410, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("150 Million",310,400);
+		words["text"] = "150 Million";
+    words["x"] = 310;
+    words["y"] = 400;
 	}
-	
+
 	var check = collidePointCircle(mouseX,mouseY, 517, 360, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("200 Million",500,345);
+		words["text"] = "200 Million";
+    words["x"] = 500;
+    words["y"] = 345;
 	}
-	
+
 	var check = collidePointCircle(mouseX,mouseY, 751, 260, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("300 Million",730,245);
+		words["text"] = "300 Million";
+    words["x"] = 730;
+    words["y"] = 245;
 	}
-	
+
 		var check = collidePointCircle(mouseX,mouseY, 889, 160, 10, 10); //see if the mouse is in the rect
 
 	if(check){ //change color!
 	  noStroke;
-		text("400 Million",860,150);
+		words["text"] = "400 Million";
+    words["x"] = 860;
+    words["y"] = 150;
 	}
-	
+
+  text(words["text"], words["x"], words["y"])
```

Comment:
Part of the assignment was specifically to make this in an object oriented manner. See https://github.com/ArtG2260SpringTuesday2016/class9/blob/master/sketch.js